Logger/BaseLogger.cs

```diff
@@ -2,6 +2,9 @@
 {
     public abstract class BaseLogger
     {
+        //Auto property of className which is set in logFactory
+        public string className { get; set; }
```

Comment:
Properties should be named with PascalCase
```suggestion
        public string ClassName { get; set; }
```